drivers/hmis/app/models/hmis/hud/client.rb

```diff
@@ -167,6 +168,22 @@ def mci_id
     ac_hmis_mci_ids.to_a.min_by(&:id)&.value
   end
 
+  def update_mci_client
```

Comment:
Hmm we should only do this on form submission, not every time the client is saved. I don't think there is a "save context" like there is for validation contexts, so I think the only way is to set an attribute like we have above.

```
    # Post-save action to create a new MCI ID if specified by the ClientProcessor
    if update_mci_attributes && HmisExternalApis::AcHmis::Mci.enabled?
      self.update_mci_attributes = nil
      # run job if relevant attributes changed
    end
```

Then in the ClientProcessor, we should set that attribute in [this case](https://github.com/greenriver/hmis-warehouse/blob/5d70e6ce358172f23c9fc205472e1d8914106b15/drivers/hmis/app/models/hmis/hud/processors/client_processor.rb#L160-L161) and [if we get here](https://github.com/greenriver/hmis-warehouse/blob/5d70e6ce358172f23c9fc205472e1d8914106b15/drivers/hmis/app/models/hmis/hud/processors/client_processor.rb#L184).

Does that make sense?

Reply:
Do you mean that it should be set to `false` in the former case and `true` in the latter? That makes sense if so. It that's not what you mean, can you please explain in more detail?

Reply:
In both cases I linked, it should be set to "true".

The first case is when someone is submitting a form for a client that has an MCI ID. When that happens, we want to hit their API if any of the relevant fields have changed, so that they are aware of any new data associated with that ID.
The second case is when someone is linking a client to an MCI ID for the first time. When that happens, we want to hit their API with the relevant fields, so that they are aware of any new data associated to that ID. It's fine to only make the call if any of the MCI-relevant fields have `changed?` because they will have changed if the MCI ID is added.